File: app/src/main/java/com/chess/cryptobot/model/response/HistoryResponseFactory.java
package com.chess.cryptobot.model.response;

import com.chess.cryptobot.model.History;

import java.util.ArrayList;
import java.util.List;

public class HistoryResponseFactory {
    private final List<? extends HistoryResponse> historyResponseList;

    public HistoryResponseFactory(List<? extends HistoryResponse> historyResponseList) {
        this.historyResponseList = historyResponseList;
    }

    public List<History> getHistory() {
        List<History> historyList = new ArrayList<>();
        for (HistoryResponse historyResponse : historyResponseList) {
            History history = new History();
            history.setAction(historyResponse.getHistoryAction());
            history.setAmount(historyResponse.getHistoryAmount());
            history.setCurrencyName(historyResponse.getHistoryName());
            history.setDateTime(historyResponse.getHistoryTime());
            history.setMarket(historyResponse.getHistoryMarket());
            history.setPrice(historyResponse.getHistoryPrice());
            history.setProgress(historyResponse.getProgress());
            historyList.add(history);
        }
        return historyList;
    }
}

File: app/src/main/java/com/chess/cryptobot/model/response/HistoryResponseFactory.kt
package com.chess.cryptobot.model.response

import com.chess.cryptobot.model.History
import java.time.ZoneId
import java.util.*

class HistoryResponseFactory(private val historyResponseList: List<HistoryResponse>?) {
    val history: List<History>
        get() {
            val historyList: MutableList<History> = ArrayList()
            if (historyResponseList==null) return historyList
            for (historyResponse in historyResponseList) {

                val history =  History()
                history.action = historyResponse.historyAction
                history.amount = historyResponse.historyAmount
                history.currencyName = historyResponse.historyName
                history.dateTime = historyResponse.historyTime?.withZoneSameInstant(ZoneId.systemDefault())?.toLocalDateTime()
                history.market = historyResponse.historyMarket
                history.price = historyResponse.historyPrice
                history.progress = historyResponse.progress
                historyList.add(history)
            }
            return historyList
        }

}